tensorflow/c/experimental/pluggable_device_utils/pluggable_device_utils.h

```diff
@@ -0,0 +1,48 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#ifndef TENSORFLOW_C_EXPERIMENTAL_PLUGGABLE_DEVICE_UTILS_PLUGGABLE_DEVICE_UTILS_H_
+#define TENSORFLOW_C_EXPERIMENTAL_PLUGGABLE_DEVICE_UTILS_PLUGGABLE_DEVICE_UTILS_H_
+
+#include "tensorflow/c/tf_status_helper.h"
+#include "tensorflow/core/platform/status.h"
+
+namespace pluggable_device {
+
+#define VALIDATE_STRUCT_SIZE(STRUCT_NAME, STRUCT_OBJ, SIZE_VALUE_NAME)    \
+  do {                                                                    \
```

Comment:
@penpornk The VALIDATA_STRUCT_SIZE seems be moved to [c_api_macros.h](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/c/c_api_macros.h#L19-L21) after this PR merged, this will cause build failure in plugin. VALIDATE_STRUCT_SIZE is not a public C API(only used by stream_executor.cc and pluggable_profiler.cc). it contains some C++ struct(tensorflow::Status), so it will cause some issue when c_api_macros.h is included by plugin, I think this part of code needs to be moved to an TensorFlow internal files. from c_api_macros.h, Thanks very much.
build errror:
```
bazel-out/k8-opt/bin/external/local_config_tf/include/tensorflow/core/platform/status.h:287:65: note: passing argument to parameter 'v' here
inline tensorflow::string* TfCheckOpHelper(::tensorflow::Status v,
^
itex/core/graph/onednn_layout/onednn_layout.cc:263:15: error: no viable conversion from 'itex::Status' to '::tensorflow::Status'
TF_CHECK_OK(GetNodeAttr(node_def, "transpose_b", &trans_b));
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
It doesn't look related. Looks like you are calling TF_CHECK_OK, which is a TF macro that expands into a function (TfCheckOpHelper) that expects ::tensorflow::Status. GetNodeAttr should return ::tensorflow::Status but somehow the compiler thinks it is itex::Status. Note that VALIDATE_STRUCT_SIZE has always depended on ::tensorflow::Status. We added ifdef __cplusplus to ensure inclusion of c_api_macros into C code is OK.

Reply:
Thanks for the help! we find that we have duplicate name of macros in c_api_macros.h, we renamed them and it works successfully.